custom/nphcda/management/commands/update_cases_fully_vaccinated.py

```diff
@@ -0,0 +1,167 @@
+    if the case property count_of_selected_antigens >= total count of eligible vaccines then 1, else 0
+    """
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--limit',
+            type=int,
+            help='Limit cases'
+        )
+        parser.add_argument(
+            '--real-run',
+            action='store_true',
+            default=False,
+            help='Do real updates'
+        )
+
+    def handle(self, *args, **options):
+        self.real_run = options.get('real_run')
+        self.limit = options.get('limit')
+
+        if self.limit:
+            logger.info(f"You have requested to limit cases to {self.limit}")
+
+        if self.real_run:
+            logger.warning("You are doing a real update, please confirm before proceeding.")
+            confirmation = input("Y/N?")
+            if confirmation != 'Y':
+                logger.info("Aborting!")
+                exit(0)
+            else:
+                logger.warning("Proceeding!")
+        else:
+            logger.info("This is a demo run so no real updates will be made.")
+
+        self._update()
+
+        logger.info("Completed. Bye!")
+
+    def _update(self):
+        query = self._get_es_query()
+        expected_cases_to_be_iterated = query.count()
+
+        logger.info(f"Expected cases to be iterated: {expected_cases_to_be_iterated}")
+
+        self._update_cases(query)
+
+    def _update_cases(self, query):
+        total_count_of_cases_with_updates = 0
+
+        doc_ids = with_progress_bar(
+            query.get_ids(),
+            query.count()
+        )
+        for chunk in chunked(doc_ids, CHUNK_SIZE, list):
+            count_of_cases_with_updates = self._update_cases_chunk(chunk)
+            logger.info(f"Cases with updates: {count_of_cases_with_updates}")
+            total_count_of_cases_with_updates += count_of_cases_with_updates
+
+        logger.info(f"Total cases with updates: {total_count_of_cases_with_updates}")
+
+    def _get_es_query(self):
+        search_string = "household_member_type = 'under_5' and age_in_months < 60 and fully_vaccinated =''"
+        query = (
+            CaseSearchES().
+            domain(DOMAIN).
+            case_type(HOUSEHOLD_MEMBER_CASE_TYPE).
+            is_closed(False)
+        )
+        if self.limit:
+            query = query.size(self.limit)
+        return query.xpath_query(DOMAIN, search_string)
+
+    def _update_cases_chunk(self, case_ids):
+        cases = CommCareCase.objects.get_cases(case_ids)
+        case_updates = {}
+        for case in cases:
+            _updates = self._case_property_updates_for_case(case)
+            if _updates:
+                if not self.real_run:
+                    logger.info(f"{case.case_id}: {_updates}")
+                case_updates[case.case_id] = _updates
+        if self.real_run:
+            try:
+                xform, cases = bulk_update_cases(DOMAIN, case_updates, self.__module__)
```

Comment:
It looks like `case_updates` is a dictionary. But `bulk_update_cases()` [takes a tuple](https://github.com/dimagi/commcare-hq/blob/034a41ff314457e5a5a12052acd28953cfadc9ff/corehq/apps/hqcase/utils.py#L225).

(Damn, _if only_ Python had a way to indicate expected parameter types to developers, that our IDEs could parse, and warn us when we're passing a different type! :trollface: :roll_eyes: :expressionless: )

Reply:
Thanks for catching that @kaapstorm 
https://github.com/dimagi/commcare-hq/pull/36461/commits/b605bb5fdc570cec4c92c8bdd743149b89799801